rscp/client.go

```diff
@@ -105,17 +102,15 @@ func (c *Client) receive() ([]Message, error) {
 // connect create connection
 func (c *Client) connect() error {
 	Log.Infof("Connecting to %s", c.connectionString)
-	var (
-		conn net.Conn
-		err  error
-	)
-	if conn, err = net.DialTimeout("tcp", c.connectionString, c.config.ConnectionTimeout); err != nil {
-		c.isConnected = false
+
+	conn, err := net.DialTimeout("tcp", c.connectionString, c.config.ConnectionTimeout)
+	if err != nil {
 		return err
 	}
-	c.conn = conn
-	c.isConnected = true
+
 	Log.Infof("successfully connected to %s", c.conn.RemoteAddr())
+	c.conn = conn
```

Comment:
Needs to be above line 111, as `c.conn` is accessed when `nil`.

Reply:
Tested with the the above mentioned change and at least the command line utility does work as I expect it.
Would be nice if you could squash the commits into one, so I could do a single commit rebase. Otherwise I would do a squash merge.

Reply:
Thanks, fixed. Would be great if you could squash on merge.